operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -0,0 +1,190 @@
+        return keycloak.getMetadata().getName() + "-update-job";
+    }
+
+    private static String podName(Keycloak keycloak) {
+        return keycloak.getMetadata().getName() + "-update-pod";
+    }
+
+    private static ObjectMeta createMetadata(String name, Keycloak keycloak) {
+        var builder = new ObjectMetaBuilder();
+        builder.withName(name)
+                .withNamespace(keycloak.getMetadata().getNamespace())
+                .withLabels(Utils.allInstanceLabels(keycloak))
+                .withAnnotations(Map.of(KEYCLOAK_CR_HASH_ANNOTATION, keycloakHash(keycloak)));
+        return builder.build();
+    }
+
+    private static void addPodSpecTemplate(JobSpecFluent<?> builder, Keycloak keycloak, Context<Keycloak> context) {
+        var podTemplate = builder.withNewTemplate();
+        podTemplate.withMetadata(createMetadata(podName(keycloak), keycloak));
+        podTemplate.withSpec(createPodSpec(context));
+        podTemplate.endTemplate();
+    }
+
+    private static PodSpec createPodSpec(Context<Keycloak> context) {
+        var builder = new PodSpecBuilder();
+        builder.withRestartPolicy("Never");
+        addInitContainer(builder, context);
+        addContainer(builder, context);
+        builder.addNewVolume()
+                .withName(WORK_DIR_VOLUME_NAME)
+                .withNewEmptyDir()
+                .endEmptyDir()
+                .endVolume();
+        // For test KeycloakDeploymentTest#testDeploymentDurability
+        // it uses a pause image, which never ends.
+        // After 60 seconds, the job is terminated allowing the test to complete.
+        // TODO should be configurable?
+        builder.withActiveDeadlineSeconds(POD_DEADLINE_SECONDS);
+        return builder.build();
+    }
+
+    private static void addInitContainer(PodSpecBuilder builder, Context<Keycloak> context) {
+        var existing = CRDUtils.firstContainerOf(ContextUtils.getCurrentStatefulSet(context)).orElseThrow();
+        var containerBuilder = builder.addNewInitContainerLike(existing);
+        configureContainer(containerBuilder, INIT_CONTAINER_NAME, INIT_CONTAINER_ARGS);
+        containerBuilder.endInitContainer();
+    }
+
+    private static void addContainer(PodSpecBuilder builder, Context<Keycloak> context) {
+        var existing = CRDUtils.firstContainerOf(ContextUtils.getDesiredStatefulSet(context)).orElseThrow();
+        var containerBuilder = builder.addNewContainerLike(existing);
+        configureContainer(containerBuilder, CONTAINER_NAME, CONTAINER_ARGS);
+        containerBuilder.endContainer();
+    }
+
+    private static void configureContainer(ContainerFluent<?> containerBuilder, String name, List<String> args) {
+        containerBuilder.withName(name);
+        containerBuilder.withArgs(replaceStartWithUpdateCommand(containerBuilder.getArgs(), args));
+        // remove volumes, won't be used
+        containerBuilder.withVolumeDevices();
+        containerBuilder.withVolumeMounts();
```

Comment:
People might use them for one thing or another (maybe to mount extensions or others), so I would not touch volumes. Maybe the configs points to them, so we would soon re-add them if we remove them now.

Reply:
Volumes took me a while to get the configuration right. Let me know what you think of the new code.